### PR TITLE
861: Developed context actions group, added generate routes.xml context action

### PR DESCRIPTION
**Pull request by bohdan-harniuk**
**Description** (*)

Developed abstraction for quick adding new context dependent file generation actions.
To add new file generation action that depends on context:

1. create an action class that extends `com.magento.idea.magento2plugin.actions.context.AbstractContextAction`
2. register it in the plugin.xml file inside the **MagentoContextBasedActionsGroup** group
3. implement the **isVisible** method to define condition where to show this action
4. implement the **getProperties** method to specify properties that should be auto-populated from the known context (leave empty for user-defined values)

Added new **routes.xml** generation action.

Conditions to show this action:

1. an etc/adminhtml/frontend folder inside the Magento 2 module

Auto-populated variables from the context:

1. MODULE_NAME
2. ROUTER_ID

Screenshot 2022-02-05 at 16 06 56

Screenshot 2022-02-05 at 16 07 41

Screenshot 2022-02-05 at 16 07 56

Screenshot 2022-02-05 at 16 08 16

**Fixed Issues (if relevant)**

1. Fixes magento/magento2-phpstorm-plugin#861

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: resources/fileTemplates/internal/Magento Routes XML.xml.html [hunk 1]
```diff
@@ -30,14 +30,43 @@
...
+    <tr>
+        <td valign="top"><nobr><font face="verdana" size="-2"><b>${MODULE_NAME}</b></font></nobr></td>
+        <td width="10">&nbsp;</td>
+        <td width="100%" valign="top"><font face="verdana" size="-1">Specifies the name of your module
+        </font>
+        </td>
+    </tr>
 </table>
 </body>
 </html>
```
Comment (andrewbess) [inline resources/fileTemplates/internal/Magento Routes XML.xml.html:72]: Could you please add new line at end of file

**Review by andrewbess — CHANGES_REQUESTED** [review]
Hello @bohdan-harniuk 
Thank you for your contribution.
Please check my comment.
Thank you in advance

File: resources/fileTemplates/internal/Magento Routes XML.xml.html (hunk 1 above)
Comment (bohdan-harniuk) [reply resources/fileTemplates/internal/Magento Routes XML.xml.html:72]: Done

**Comment by bohdan-harniuk** [comment]
Hello, @andrewbess!

Thank you for your review!
I've fixed what you mentioned in the review.

Regards,

**Review by andrewbess — APPROVED** [review]
Hello @bohdan-harniuk 
Thank you for the fast fix
Approved ✔️

**Pull request merged**